WIP: LR and MLP models for vehicles.csv, please review evaluation methods, unsure about required plots

namrathagopalabhatla opened the pull request:
I am unsure about what plots I should include for the evaluation metrics. Please advise.

namrathagopalabhatla commented:
I am unsure about what plots I should include for the evaluation metrics. Please advise.

dzeber commented on dev/namrathagopalabhatla/classification_models.py:
```diff
@@ -0,0 +1,148 @@
+class MLP:
+    def __init__(self, hidden_size=None, activation='relu', solver='lbfgs', max_iter=1000):
+        '''Constructor for initalization of model using provided parameters'''
+        super(MLP, self).__init__()
```
This (and line 86) is not needed if your class is not inheriting explicitly from another one.

namrathagopalabhatla commented:
@dzeber Thank you so much for your suggestions. I will look into it. I am sorry i have been inactive for the past few days. There are a lot of issues that came up with network issues due to the whole pandemic situation.

namrathagopalabhatla commented:
I have updated the code as per your suggestions. I have also added new code for Issue #3 in my latest commit, where I have added code for AUROC values and ROC curves, and a plot of TP and FP marked separately for each class of the classificarion problem, to visualize the misclassifications in 2D using LDA. Kindly review.

namrathagopalabhatla commented:
I have updated the repo with the fixes suggested by @dzeber in my first pull request that worked on the startup task #2. I will also add the Eigen value spectrum visualization (Scree plot) in the subsequent commits that also include work on other issues. Kindly review.

namrathagopalabhatla commented:
I have completed updates according to all the suggestions. I have also incorporated the train-test splits for #3 . Kindly review and tell me if the code for #3 needs anything else

dzeber commented:
Thanks, your notebook for issue 2 is much improved! It is interesting to see those plots from the PCA - helps explain why those two classes (saab and opel I think) are so difficult to distinguish. Also, good work on the issue 3 notebook - it is very well executed.
